providers/cloudflare/cloudflareProvider.go

```diff
@@ -458,6 +459,17 @@ func newCloudflare(m map[string]string, metadata json.RawMessage) (providers.DNS
 		return nil, fmt.Errorf("if cloudflare apitoken is set, apikey and apiuser should not be provided")
 	}
 
+	var err error
+	if api.APIToken != "" {
+		api.cfClient, err = cloudflare.NewWithAPIToken(api.APIToken)
+	} else {
+		api.cfClient, err = cloudflare.New(api.APIKey, api.APIUser)
+	}
+
+	if err != nil {
+		return nil, fmt.Errorf("cloudflare did not like whatever you did %s", err)
```

Comment:
How about:

```
return nil, fmt.Errorf("cloudflare credentials: %w", err)
```

Reply:
OMG, I thought I changed that. That’s embarrassing…